src/parser/mod.rs

```diff
@@ -7295,6 +7295,29 @@ impl<'a> Parser<'a> {
         }
     }
 
+    pub fn parse_number_value_with_sign(&mut self) -> Result<Expr, ParserError> {
+        let next_token = self.next_token();
+        match next_token.token {
+            tok @ Token::Minus | tok @ Token::Plus => {
+                if tok == Token::Plus {
+                    Ok(Expr::UnaryOp {
+                        op: UnaryOperator::Plus,
+                        expr: Box::new(Expr::Value(self.parse_number_value()?)),
+                    })
+                } else {
+                    Ok(Expr::UnaryOp {
+                        op: UnaryOperator::Minus,
+                        expr: Box::new(Expr::Value(self.parse_number_value()?)),
+                    })
+                }
+            }
```

Comment:
```suggestion
            Token::Minus => {
                Ok(Expr::UnaryOp {
                    op: UnaryOperator::Minus,
                    expr: Box::new(Expr::Value(self.parse_number_value()?)),
                })
            }
            Token::Plus => {
                Ok(Expr::UnaryOp {
                    op: UnaryOperator::Plus,
                    expr: Box::new(Expr::Value(self.parse_number_value()?)),
                })
            }
```
thinking we can avoid the double work of match+comparison?